src/BinCommand.php

```diff
@@ -33,57 +34,85 @@ protected function configure()
      */
     public function execute(InputInterface $input, OutputInterface $output)
     {
-        $this->resetComposers();
+        $this->resetComposers($application = $this->getApplication());
+        /** @var ComposerApplication $application */
 
         putenv('COMPOSER_BIN_DIR='.$this->createConfig()->get('bin-dir'));
 
-        $binVendorRoot = 'vendor-bin';
+        $binvendorRoot = 'vendor-bin';
         $binNamespace = $input->getArgument('namespace');
         $input = new StringInput(preg_replace('/bin\s+' . preg_quote($binNamespace, '/') . '/', '', $input->__toString(), 1));
 
-        if ('all' !== $binNamespace) {
-            $binRoot = $binVendorRoot.'/'.$binNamespace;
-            if (!file_exists($binRoot)) {
-                mkdir($binRoot, 0777, true);
-            }
-
-            $this->chdir($binRoot);
-
-            return $this->getApplication()->doRun($input, $output);
-        }
+        return ('all' !== $binNamespace)
```

Comment:
gist of the refactor: I find it easier to read in the sense that it conveys more what it does